### DISCUSSION TITLE
开发者体验的基础设施 | 夜天之书

### DISCUSSION THREAD

**Discussion by giscus** [tisonkun/blog#49] [Comments]
# 开发者体验的基础设施 | 夜天之书

本文演绎自 Kenneth Auchenberg 的 Developer Experience Infrastructure (DXI) 博客。 原文提到，现在大部分号称以开发者优先为理念建设基础设施的公司，对开发者体验的定义和落地都不是最佳实践。原文作者认为，随着业界对开发者体验的重视和发展，将会出现一个新的领域 Developer Experience Infrastructure (DXI)

https://www.tisonkun.org/2022/10/28/developer-experience-infrastructure/

**Comment by ZoomQuiet** [comment]
#是也乎,(￣▽￣)
之所以是门艺术, 原因就在:

- 虽然当前项目是由自己和小伙伴们从0开始构建出来的
- 但是,在穿过所有困难之后, 其实大家并不能立即回想起当初自己作为自己项目最初用户时最初期望是什么
- DXI 构建的核心线索就是:

> 如何令一位有兴趣的开发者,最短时间里理解工程目标,并能正确跑起来一个最小完备demo

说穿了, 也就是说: 如果将一个开源项目当成一位虚拟偶像来打造时,
应该在哪些界面上初始化哪些内容, 并针对不同界面的最大人群优化到不用艰难探索就可以上手?

往根里说, 其实是内心第一渴望的排名:

- 自己和团队成员可以顺利开展维护是第一位的?
- 还是,项目能吸引越来越多开发者加入是第一位的?

表现在具体成本上就是:

- 团队成本核算时, 是否愿意将完善 DXI 的时间也计入普通日常工作量里来?

**Reply by tisonkun** [reply]
> * 自己和团队成员可以顺利开展维护是第一位的？
> * 还是，项目能吸引越来越多开发者加入是第一位的？

这两者虽然在某些情况下可能不能两全，但是大部分情况下应该还是可以兼顾的。我个人的体验是，如果参照 Linus 的思路，只跟自己愿意与之合作的人合作，而不是刻意追求越来越”多“的开发者加入，这两者并不冲突。

我所看到的良性发展的开源社群大多是志同道合的人建设起来的，而且往往有非常强劲的生产力核心。这样才不会在早期的时候就凋敝或者被分裂。

**Reply by tisonkun** [reply]
成本核算的部分，我目前还没能找到特别直观的指标来衡量 DXI 的关键指标和产出。典型的比如 time to hello world 或者用户报 bug 能不能带上有效的报告，目前都是非常主观和定性的。让我来评估和分析，我当然能够罗列出来问题和解决方案，但是很难定量的给出关键指标或者规模化。

**Reply by ZoomQuiet** [reply]
#是也乎,(￣▽￣)

其实第一个问题, 本质就指向这个社区算谁的, 自己的, 当然以志同道合的伙伴们为先,
公司的, 那就看 KPI 制定什么了...

这可能才是为什么企业发起的开源项目总是很奇怪的原因, 当然, 中国企业, 国外的对这种一时看不出收益点的,
能真的置之不理, 放心生长...这就不简单了.

成本的部分 , 是否可以参考 微软的文档工程师怎么衡量的?
毕竟, 异步离线构建 DXI 体系时, 多数还是良好文档的积累...

**Reply by tisonkun** [reply]
> 公司的, 那就看 KPI 制定什么了...

哎，这个其实很难。没有一个 owner 对项目倾注热情，依靠公司资本来建设社群，是没有灵魂的。哪怕是公司运作的成功社群，很多也是创始人定下的基调和目标。连项目都是公司摊派下来的工作，做社群怎么能有斗志呢？

> 成本的部分 , 是否可以参考微软的文档工程师怎么衡量的?

有相关链接吗？我想学习一个。

**Reply by ZoomQuiet** [reply]
> A: 微软的文档工程师主要通过以下几个维度来衡量和评估日常工作量和成本:

1. 文档数量和规模:包括需要编写和维护的文档数,每个文档的页数或字数等。更多和更长的文档意味着更大的工作量。
2. 文档类型和复杂度:参考文档、教程、API文档等类型的工作量和难度各不相同。而且文档主题的艰深程度也是重要考量因素。
3. 翻译需求:对不同语言的文档翻译也是文档工程师工作的一部分,需要评估各语言文档的数量和难易程度。
4. 更新频率:文档内容更新频繁意味着维护工作量较大,这也是一个重要标准。
5. 生命周期管理:文档的孵化到废止的整个生命周期管理也需要投入大量时间,这也应在成本评估中考虑。
6. 工具和流程:采用的文档管理工具、编写工具的复杂性,以及文档流程的规范性都会对工作效率产生影响,这也是工作量评估的因素之一。

除此之外,时间追踪和工作报告是了解工作总体情况的重要方式。微软还会基于具体项目组或部门的情况制定相应的工作量评估标准和流程。

以上 Claude 的回答, 没找到出处...
